test/modules/TestResultCoordinator/Reports/CountingReport.cs

```diff
@@ -19,7 +19,8 @@ class CountingReport : TestResultReportBase
             ulong totalExpectCount,
             ulong totalMatchCount,
             ulong totalDuplicateResultCount,
-            IReadOnlyList<TestOperationResult> unmatchedResults)
+            IReadOnlyList<TestOperationResult> unmatchedResults,
+            Option<bool> stillReceivingFromEventHub)
```

Comment:
Could the `stillReceivingFromEventHub` be a default value? Looks like most of the cases in this pr are the same. Also, does this need to be optional if it is a boolean? As with option, we will have 3 states: Some(true), Some(false), None. I am sure you picked this for a reason, would you mind explaining to me what each state is representing? would help me a lot! :)

Reply:
Making it an option here gives us a benefit of having the 3 states: true, false, and N/A. If it is None, then we are explicitly saying that we can't have a true or a false in this situation. I added a comment to make this more clear